File: Backend/src/main/java/com/vivek/User/Management/System/service/UserService.java
package com.vivek.User.Management.System.service;


import com.vivek.User.Management.System.model.User;
import com.vivek.User.Management.System.repository.UserRepository;
import org.springframework.stereotype.Service;
import org.springframework.web.bind.annotation.PathVariable;

import java.util.List;
import java.util.Optional;

@Service
public class UserService {

    UserRepository userRepository;

    public UserService(UserRepository userRepository){
        this.userRepository=userRepository;
    }

    public List<User> getAllUsers(){
        return userRepository.findAll();
    }

    public User createUser(User user){
        return userRepository.save(user);
    }

    public User findByUserId(Long id){
        return userRepository.findById(id).orElseThrow();
    }

    public Optional<User> findByPhoneNumber(String phoneNumber){
        return userRepository.findByPhoneNumber(phoneNumber);
    }

    public List<User> findRegisteredTodayUser(){
        return userRepository.findUserRegisteredToday();
    }
    public Optional<User> findByName(String name){
        return userRepository.findByName(name);
    }
    public User updateUser(User user,Long id){
     User existingUser= userRepository.findById(id).orElseThrow();
     existingUser.setName(user.getName());
     existingUser.setEmail(user.getEmail());
     existingUser.setPhoneNumber(user.getPhoneNumber());
     return userRepository.save(existingUser);
    }

    public void deleteUser(Long id){

        userRepository.deleteById(id);
    }

}

File: frontend/src/services/userService.js
import axios from 'axios';

const BASE_URL = 'http://localhost:8080/api';

export const getAllUsers = () => axios.get(`${BASE_URL}/users`);
export const getUserById = (id) => axios.get(`${BASE_URL}/user/${id}`);
export const createUser = (user) => axios.post(`${BASE_URL}/user`, user);
export const updateUser = (id, user) => axios.put(`${BASE_URL}/user/${id}`, user);
export const deleteUser = (id) => axios.delete(`${BASE_URL}/user/${id}`);
export const findByPhoneNumber = (phoneNumber) => axios.get(`${BASE_URL}/user/phoneNumber/${phoneNumber}`);
export const findByName = (name) => axios.get(`${BASE_URL}/user/name/${name}`);
export const getTodayRegisteredUsers = () => axios.get(`${BASE_URL}/user/registered-today`);
